### PR TITLE
Add gitpod support for puppet modules

### PR DESCRIPTION
**Pull request by logicminds**
* This adds initial support for gitpod puppet development.  Any user
    who opens the module in gitpod would automatically
    stand up a PDK and puppet development environment.

    While this is a first stab at this file I expect further customizations
    to follow and allow the user to utilize the sync or config_defaults
    to tweak their own unique gitpod.

    I decided to leave the clickable banner that normally goes in the readme out of this changeset as I don't know
    what the user's gitpod link would actually be.  I am sure a config variable might help with this but decided to KISS.

An example of this setup can be viewed here by clicking this link: [Gitpod ready-to-code](https://gitpod.io/#https://gitlab.com/nwops/control-repo.git)

### CODE REVIEW

**Comment by michaeltlombardi** [comment]
We're discussing this one internally a bit, as it will effect _all_ Puppet modules. :)

**Comment by logicminds** [comment]
Yes similar to Travis, appveyor and other stuff people ignore if they don't use.

**Review by DavidS — CHANGES_REQUESTED** [review]
I'm happy to merge this, if we tag this as `unmanaged: true` in the config defaults. That way the functionality is there for people who want it.

> other stuff people ignore if they don't use
We've heard enough complaints about the confusing maze of files we deploy for newcomers that I don't want to add on top of that.
I'd disable all non-essential stuff too for a pdk 2.0 release.

@binford2k once we get this merged, I suggest we enable it on all supported modules to see if that brings in more/better contributions.

**Comment by logicminds** [comment]
@DavidS made the changes as requested and updated the README as well to show how to enable.

**Review by DavidS — APPROVED** [review]

**Pull request merged**
